### PR TITLE
feat: support media channels

### PR DESCRIPTION
**Pull request by onerandomusername**
## Summary

~~Untested, I don't have access to media channels.~~

Tested now.

## Checklist

- [x] If code changes were made, then they have been tested
    - [x] I have updated the documentation to reflect the changes
    - [x] I have formatted the code properly by running `pdm lint`
    - [x] I have type-checked the code by running `pdm pyright`
- [ ] This PR fixes an issue
- [ ] This PR adds something new (e.g. new method or parameters)
- [ ] This PR is a breaking change (e.g. methods or parameters removed/renamed)
- [ ] This PR is **not** a code change (e.g. documentation, README, ...)

### CODE REVIEW

File: disnake/channel.py [hunk 1]
```diff
@@ -100,7 +102,87 @@ async def _single_delete_strategy(messages: Iterable[Message]) -> None:
         await m.delete()
 
 
-class TextChannel(disnake.abc.Messageable, disnake.abc.GuildChannel, Hashable):
+class WebhookableChannel:
```
Comment (shiftinv) [inline disnake/channel.py:105]: this should be a separate PR

File: disnake/flags.py [hunk 2]
```diff
@@ -2054,27 +2054,44 @@ class ChannelFlags(BaseFlags):
...
-        This only applies to channels of type :class:`ForumChannel`.
+        This only applies to channels of types :class:`ForumChannel` and :class:`MediaChannel`.
 
         .. versionadded:: 2.6
         """
         return 1 << 4
 
+    @flag_value
+    def hide_media_download_options(self):
+        """:class:`bool`: Returns ``True`` if the channel requires all newly created threads to have a tag.
```
Comment (Victorsitou) [inline disnake/flags.py:2087]: wrong docstring :p

File: disnake/state.py [hunk 3]
```diff
@@ -1114,8 +1115,8 @@ def parse_thread_create(self, data: gateway.ThreadCreateEvent) -> None:
         guild._add_thread(thread)
         if not has_thread:
             if data.get("newly_created"):
-                if isinstance(thread.parent, ForumChannel):
-                    thread.parent.last_thread_id = thread.id
+                if type(thread.parent) in (ForumChannel, MediaChannel):
```
Comment (Victorsitou) [inline disnake/state.py:1118]: why this over an `isinstance`?

File: disnake/webhook/async_.py [hunk 4]
```diff
@@ -975,13 +975,15 @@ def guild(self) -> Optional[Guild]:
...
+    def channel(
+        self,
+    ) -> Optional[Union[TextChannel, VoiceChannel, StageChannel, ForumChannel, MediaChannel]]:
+        """Optional[Union[:class:`TextChannel`, :class:`VoiceChannel`, :class:`StageChannel`, :class:`ForumChannel`, :class:`MediaChannel`]]: The channel this webhook belongs to.
 
         If this is a partial webhook, then this will always return ``None``.
 
-        Webhooks in :class:`ForumChannel`\\s can not send messages directly,
-        they can only create new threads (see ``thread_name`` for :attr:`Webhook.send`)
+        Webhooks in channels of types :class:`ForumChannel' or :class:`MediaChannel` can not send messages directly.
```
Comment (Victorsitou) [inline disnake/webhook/async_.py:985]: the UI doesn't seem to allow the creation of webhooks 🤔

File: disnake/state.py (hunk 3 above)
Comment (onerandomusername) [reply disnake/state.py:1118]: consistency as of how its used elsewhere

File: disnake/webhook/async_.py (hunk 4 above)
Comment (onerandomusername) [reply disnake/webhook/async_.py:985]: asked in ddevs https://canary.discord.com/channels/613425648685547541/697489244649816084/1120071900954108084

File: disnake/flags.py (hunk 2 above)
Comment (onerandomusername) [reply disnake/flags.py]: [`aa13cd1` (#1050)](https://github.com/DisnakeDev/disnake/pull/1050/commits/aa13cd18231e8d30ac23f3045151079618bab1ba)

File: disnake/webhook/async_.py (hunk 4 above)
Comment (onerandomusername) [reply disnake/webhook/async_.py]: and on the github issue https://github.com/discord/discord-api-docs/pull/6232#issuecomment-1596245447

Comment (shiftinv) [reply disnake/webhook/async_.py]: this was fixed yesterday, webhooks are now shown in the UI as well

File: disnake/channel.py (hunk 1 above)
Comment (shiftinv) [reply disnake/channel.py]: reverted in 3e5ed18f8d5c4044da4fd0f1ee41282d1cf81c6c for now. Still, refactors like that would be great as new PRs
